runtime/j9vm/javanextvmi.cpp

```diff
@@ -326,34 +327,42 @@ JVM_VirtualThreadMountEnd(JNIEnv *env, jobject thread, jboolean firstMount)
 				J9VMJDKINTERNALVMCONTINUATION_VMREF(currentThread, continuationObj));
 	}
 
+	/* On some occassions, J9AllocateObject acquires exclusive VM access. It is invoked without acquiring
+	 * liveVirtualThreadListMutex to prevent a deadlock.
+	 */
+	if (NULL == vm->liveVirtualThreadList) {
+		Assert_SC_true(firstMount);
+		J9Class *virtualThreadClass = J9OBJECT_CLAZZ(currentThread, J9_JNI_UNWRAP_REFERENCE(thread));
+		J9MemoryManagerFunctions *mmFuncs = vm->memoryManagerFunctions;
+
+		/* Allocate empty virtual thread and create a global reference to it as root for the linked list.
+		 * This prevents the root reference from becoming stale if the GC moves the object.
+		 */
+		rootVirtualThread = mmFuncs->J9AllocateObject(currentThread, virtualThreadClass, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+		if (NULL == rootVirtualThread) {
+			vmFuncs->setHeapOutOfMemoryError(currentThread);
+			goto release1;
+		}
+		J9VMJAVALANGVIRTUALTHREAD_SET_STATE(currentThread, rootVirtualThread, J9VM_VIRTUALTHREAD_ROOT_NODE_STATE);
```

Comment:
@keithc-ca https://github.com/eclipse-openj9/openj9/pull/16365#discussion_r1037263435 is addressed here.